### ISSUE TITLE
Retenção de jogadores no PerfectDreams

### ISSUE THREAD

**Issue by ItsGabii** [PerfectDreams/StickyNotes#30]
## 🤔 Antes de enviar...
Para manter tudo organizado, siga estes passos antes de começar:
* Eu já confirmei que esta sugestão será útil para outros usuários.
* Eu já procurei por sugestões parecidas no repositório.
* Eu já verifiquei que não existe nenhuma outra branch ou pull request aberto com a minha sugestão ou com uma sugestão parecida.
* Eu tentei procurar por [outros repositórios do PerfectDreams](https://github.com/PerfectDreams) um repositório relacionado com a minha sugestão, mas não encontrei/é uma sugestão sobre algo que não tem nada relacionado no servidor.

### 🎮 Seu nome no PerfectDreams
*Its_Gabi*

### ✨ Descrição da sugestão
 Para fazer com que players permaneçam mais tempo logados no server eu creio que é bom promover eventos/recompensas e outros eventos que motivem os jogadores e permitam que eles nunca sintam tédio dentro do servidor. Mas primeiro, ressalto que existe a possibilidade de alguma das sugestões fugir um tanto do estilo survival do PerfectDreams, mas para evitar que isso aconteça eu achei que seria adequado ver a jornada de alguém no server como um player que primeiramente precise procurar seus recursos pra viver após isso, o players buscaria aumentar a sua quantidade de itens fazendo farms básicas, construindo vilas e etc.

**⚔Mcmmo**
* **Aviso no chat:** assim como acontece em alguns outros servers, mandar uma mensagem para todos caso um player alcance determinados níveis do mcmmo em determinada área, sugiro avisar a cada 200 níveis e também para continuar incentivando quem já tem um alto lv, talvez possam adicionar mapas.
Exemplo tosco:
![Obviamente, haverão outras cores de picareta para poder alcançar nivéis maiores](https://i.imgur.com/GbignP1.png)

* **Boost:** realizar eventos de boost de x1.5 em xp de skills para aqueles que logaram 7 dias seguidos e ficaram jogando ao menos 3 horas em cada dia. Os Boosts também poderiam acontecer em datas especiais (data de lançamento do Perfect, data de aniversário do dono, natal, entre outros).
 
* **Recompensas para top 1:** caso alguém chegue no top 1 de alguma skill pela primeira vez, dar um prêmio especial, como um item exclusivo, como um mapa ou itens que apenas podem ser criados por meio de comandos ou serem programados (as armaduras da CityCave por exemplo) apenas para servir de itens colecionáveis. 

* **Recompensas no lv 1400:** ao chegar ao lv 1400 vai ganhar... sei lá, um diamante.

**💎Eventos**
Ao contrário do Fight, Chat, Torre da morte e entre outros eventos, esses seriam organizados uma vez por mês pela staff e sempre haveria um vencedor que ganharia determinada quantia.
Para garantir que eles sejam feitos no survival, o evento precisa ter um objetivo que faça com que players gastam material, itens, xp, tempo, e outros. Por exemplo:

* **Construção:** jogadores construirão algo em específico, como enfeitar farms, fazer casas, criar decorações por tudo o mapa, entre outros (com a vantagem que o server ficaria mais bonito e talvez as pessoas vendam essas construções pra outras).

* **Esconde-esconde:** mais ou menos parecidos com eventos de procurar placas, mas um staff se esconderia em algum lugar do spawn e esperaria até alguém achar.

* **Caça ao tesouro:** do mesmo jeito que o JvGm45 tentou fazer uma vez, mas sem alguns erros e feitos com enigmas, jogos, adivinhas, etc.

**✉Mensagens automáticas para quem está fazendo nada**
Não sei se é possível detectar quando uma pessoa está fazendo nada dentro do servidor, mas, se for possível, porque não adicionar dicas do que o player fazer? Como algum tipo de farm, itens que estão em conta para ele fazer farms, avisar quando ele vai conquistar o vip de votos (se tiver) entre outros.

### 🌟 Como irá funcionar?
É meio complicado afirmar como funcionaria cada uma dessas práticas, mas creio que a maior parte do McMMo poderia ser automatizado e se não tiver como, talvez seja possível apenas avisar a staff quando alguém conquistou algo para dar o item pra ele. Os eventos seriam manuais, e as mensagens para pessoas fazendo nada seriam pré-programadas com mensagens já prontas como: "Porquê você está parado? convide alguém para ir no x1!" e programadas baseadas em acontecimentos atuais do server, como: "você sabia que falta 10 dias para você conseguir seu vip de votos?" "faltam apenas 10 níveis de escavações para chegar ao top 5!" (admito que a ideia das mensagens é um pouco difícil. 

### 😎 Porque ela seria legal no PerfectDreams?
Fazer com que players fiquem jogando por mais tempo, fazendo com que eles comprem vips assim gerando lucro para o server.

### 🍃 Qualquer outra coisa
Como eu ainda não pensei em mais maneiras para conseguir reter mais pessoas, eu provavelmente ainda irei adicionar mais coisas e reavaliar as que foram faladas aqui.

**Comment by FlammDumbFox** [comment]
O aviso no chat de níveis do mcMMO é uma ideia realmente interessante. Eu iria sugerir isso, mas não sei se o mcMMO vai estar ou não no PD quando o mesmo sair. Eu recomendaria a cada 100 níveis, já que a EXP vai (teoricamente) ser exponencial, o que complica bastante pra aumentar de nível quanto maior ele for.

Quanto aos períodos com EXP em dobro, poderíamos ter, além de datas comemorativas, nos finais de semana.

Lootboxes e ferramentas/armaduras que upam também são outras sugestões minhas que provavelmente postarei mais tarde, todas com o intuito de aumentar a retenção dos jogadores.

**Comment by FlammDumbFox** [comment]
Ah sim, e eu acho que recompensas poderiam vir a cada 200 níveis, já que o negócio aumenta de dificuldade de forma exponencial. O tempo que eu demorei pra pegar 300-700 de Excavation no Sparkly deve ter sido o que demorei pra pegar 700-800.

**Comment by MrPowerGamerBR** [comment]
Sim, anunciar os level ups do McMMO seria bem interessante (é claro, não a cada nível, e sim a cada X níveis).

> realizar eventos de boost de x1.5 em xp de skills para aqueles que logaram 7 dias seguidos e ficaram jogando ao menos 3 horas em cada dia. Os Boosts também poderiam acontecer em datas especiais (data de lançamento do Perfect, data de aniversário do dono, natal, entre outros).

Também seria legal dar recompensas (como sonhos) para quem tem um login streak no servidor.

> jogadores construirão algo em específico, como enfeitar farms, fazer casas, criar decorações por tudo o mapa, entre outros (com a vantagem que o server ficaria mais bonito e talvez as pessoas vendam essas construções pra outras).

Eu tinha pensado nisto, talvez fazer a cada um mês um evento de construção.

~~era para o ganhador ganhar uma lorota jubinha: real life 3D edition mas impressão 3D é caro~~

> mais ou menos parecidos com eventos de procurar placas, mas um staff se esconderia em algum lugar do spawn e esperaria até alguém achar.

No SparklyPower (beeeem no começo) tinha algo parecido, era um evento de esconde-esconde em que os players precisavam encontrar uma placa pelo spawn, quem encontrasse ela ganhava dinheiro.

Seria interessante fazer algo parecido no PerfectDreams. 😊 

> do mesmo jeito que o JvGm45 tentou fazer uma vez, mas sem alguns erros e feitos com enigmas, jogos, adivinhas, etc.

Se for fazer, é melhor que faça de uma maneira decente, é ruim ficar espalhando um monte de placa pelo spawn e ter que ficar explicando para o que cada coisa serve. (Já perdi a conta de quantas vezes tive que responder pessoas perguntando "porque esse item vale 1000000000 reais???")

> Não sei se é possível detectar quando uma pessoa está fazendo nada dentro do servidor, mas, se for possível, porque não adicionar dicas do que o player fazer? Como algum tipo de farm, itens que estão em conta para ele fazer farms, avisar quando ele vai conquistar o vip de votos (se tiver) entre outros.

Uma coisa que eu queria fazer é um comando que dá uma ideia aleatória para o player fazer, por exemplo...
`/ideia`

Ai aparece `Faça uma lanchonete de baixo da terra!`
